packages/icons-tnt/src/json-imports/Icons.js

```diff
@@ -1,5 +1,11 @@
-import { registerIconBundle } from "@ui5/webcomponents-base/dist/asset-registries/Icons.js";
+import { registerIconLoader } from "@ui5/webcomponents-base/dist/asset-registries/Icons.js";
 
-import SAPIcons from "../generated/assets/SAP-icons-TNT.json";
+const loadIconsBundle = async () => {
+	const iconData =  (await import("../generated/assets/SAP-icons-TNT.json")).default;
+    if (typeof iconData === "string" && iconData.endsWith(".json")) {
+        throw new Error("[icons-tnt] Invalid bundling detected - dynamic JSON imports bundled as URLs. Switch to inlining JSON files from the build or use `import \"@ui5/webcomponents-icons-tnt/dist/Assets-static.js\". Check the \"Assets\" documentation for more information.");
+    }
+    return iconData;
+}
 
-registerIconBundle("SAP-icons-TNT", SAPIcons);
+registerIconLoader("SAP-icons-TNT", loadIconsBundle);
```

Comment:
This is the part I mentioned in my previous comment - IMO we must both register the loader and run it explicitly here.

Reply:
I don't like an explicit load after the registration is a good API. This load should happen when an icon is rendered and an explicit load should be up to the app developer. Can we change this in another change, as the previous JSON behaviour was like this as well?